src/sidebars/favourites/favourite-controller.js

```diff
@@ -129,25 +131,25 @@
 
                             // Repeat the check as in the mean time a stock for this favourite could have been added.
                             if (this.stocks.map((stock1) => { return stock1.code; }).indexOf(favourite) === -1) {
-                                var data = stock.data[0],
+                                var data = stock && stock.data && stock.data[0],
```

Comment:
Do you need the first check for `stock` here? It seems below in the `else` `code` and `message` are being read off the `stock` object without a prior check, so it's assumed the object is not null/undefined.